src/ripple/app/misc/SHAMapStoreImp.cpp

```diff
@@ -566,19 +573,34 @@ SHAMapStoreImp::clearSql(
     std::string const& minQuery,
     std::string const& deleteQuery)
 {
+    assert(deleteInterval_);
     LedgerIndex min = std::numeric_limits<LedgerIndex>::max();
 
     {
-        auto db = database.checkoutDb();
         boost::optional<std::uint64_t> m;
-        *db << minQuery, soci::into(m);
+        JLOG(journal_.trace())
+            << "Begin: Look up lowest value of: " << minQuery;
+        if (auto db = database.checkoutDb())
+            *db << minQuery, soci::into(m);
+        else
+        {
+            assert(false);
+            return false;
+        }
+        JLOG(journal_.trace()) << "End: Look up lowest value of: " << minQuery;
```

Comment:
I honestly don't remember for sure why I added the `if (auto db = database.checkoutDb())` check... I think it was so that it would go out of scope and release the lock sooner, and then I added the `else` because it "wouldn't hurt". It would make more sense to rewrite it as a simple local block.